traceon/geometry.py

```diff
@@ -1134,7 +1131,180 @@ def rectangle_xy(xmin, xmax, ymin, ymax):
+            except AssertionError: pass
+
+        extruded_boundary = SurfaceCollection(extruded_boundary_paths)
+
+        if add_cap:
+            return self + extruded_boundary + self.move(*vector) 
+        else:
+            return self + extruded_boundary
+    
+    def extrude_boundary_by_path(self, path, add_cap=True):
+        """Extrude the boundary paths of a surface along a path. The path 
+        does not need to start at the surface. Imagine the  extrusion surface 
+        created by moving the boundary paths along the path.
+
+        Parameters
+        -------------------------
+        path: Path
+            The path defining the extrusion.
+        add_cap: bool
+            Whether to cap the extrusion by adding a copy of the original surface 
+            moved by the extrusion vector to the resulting SurfaceCollection.
+            
+        Returns
+        ------------------------
+        SurfaceCollection"""
+        
+        boundary = self.get_boundary_paths()
+        extruded_boundary_paths = []
+
+        for b in boundary.paths:
+            try: extruded_boundary_paths.append(b.extrude_by_path(path))
+            except AssertionError: pass
+
+        extruded_boundary = SurfaceCollection(extruded_boundary_paths)
+
+        if add_cap:
+            path_vector = path.endpoint() - path.starting_point()
+            return self + extruded_boundary + self.move(*path_vector) 
+        else:
+            return self + extruded_boundary
+    
+    def revolve_boundary_x(self, angle=2*pi, add_cap=True):
+        """Revolve the boundary paths of the surface anti-clockwise around the x-axis.
+        
+        Parameters
+        -----------------------
+        angle: float
+            The angle by which to revolve. THe default 2*pi gives a full revolution.
+        cap_extension: bool
+            Whether to cap the revolution by adding a copy of the original surface 
+            rotated over the angle to the resulting SurfaceCollection.
+
+        Returns
+        -----------------------
+        SurfaceCollection"""
+
+        boundary = self.get_boundary_paths()
+        revolved_boundary_paths = []
+
+        for b in boundary.paths:
+            try: revolved_boundary_paths.append(b.revolve_x(angle))
+            except AssertionError: pass
+
+        revolved_boundary = SurfaceCollection(revolved_boundary_paths)
+
+        if add_cap and not np.isclose(angle, 2*pi, atol=1e-8):
+            return self + revolved_boundary + self.rotate(Rx=angle)
+        else:
+            return self + revolved_boundary
+        
+    def revolve_boundary_y(self, angle=2*pi, add_cap=True):
+        """Revolve the boundary paths of the surface anti-clockwise around the y-axis.
+        
+        Parameters
+        -----------------------
+        angle: float
+            The angle by which to revolve. THe default 2*pi gives a full revolution.
+        cap_extension: bool
```

Comment:
Note that your docstrings don't match the argument names (here also I would prefer `enclose` )